modules/writing_observer/writing_observer/awe_nlp.py

```diff
@@ -202,52 +203,99 @@ async def process_texts_parallel(texts, options=None):
 
 async def process_texts(writing_data, options=None, mode=RUN_MODES.MULTIPROCESSING):
     '''
-    Process texts with caching
-    1. Create hash of text
-    2. Fetch cache data
-    3. Check to see if all options are present
-        - Yes? Add to results
-        - No? Record missing options `needed_options` and add to `need_processing`
-    4. If we need to process anything process it
-    5. Update results/cache
+    Caching:
+    1. Create text hash.
+    2. Check if hash exist in cache.
+    3. Check if some options are a subset of features_available
+        Yes: add the intersection of features_available and options to results
+    4. Check if some options are a subset of features_running.
+        Yes: a. Wait for features_running to finish.
+             b. Update the cache
+             c. Add intersection of features_running and Options to results
+    5. Check if additional features are required.
+        Yes: a. Collect options not covered till now and add to features_running.
+             b. Once finished, update cache and return results.
     '''
+    recurring_sleep_time = 1  # Time in seconds to wait between recurring calls to cache to check if features have finished running
+    running_features_wait_time = 60  # Time in seconds to wait for features already running.
```

Comment:
Parameters should either be:

-  Function parameters (with defaults)
-  Globals (and upper case)
-  In creds.yaml (with defaults)

Or some combination there-of.